### ISSUE TITLE
Next.js integration

### ISSUE THREAD

**Issue by JanKups** [chatengine-io/react-chat-engine#9]
It would be great to see Next.js supported, or if implementation steps are present in the docs.

Currently, I am receiving the following error after following the docs : "ReferenceError: WebSocket is not defined”.

**Comment by alamorre** [comment]
Nice find! I'll try to have NextJS supported this weekend.

This is a great reason to learn the NextJS framework 😁

**Comment by alamorre** [comment]
Okay so NextJS doesn't support WebSockets out of the box bc it's part server framework too. I need to install and initiate an official 3rd party w3c Socket instead. I can use this component, I'll give it a go Sunday https://www.npmjs.com/package/websocket

**Comment by JanKups** [comment]
> Okay so NextJS doesn't support WebSockets out of the box bc it's part server framework too. I need to install and initiate an official 3rd party w3c Socket instead. I can use this component, I'll give it a go Sunday https://www.npmjs.com/package/websocket

Did you have any luck with Websocket?

**Comment by alamorre** [comment]
I tried earlier in the week without luck. Let me try again tonight - also I went skiing Sunday 😋

No web-socket components on NPM are NextJS compatible under the hood. I may be down to make one bc it'll get a ton of downloads and be free visibility to ChatEngine.

All this to say I'll keep poking around on this!

**Comment by alamorre** [comment]
Done 🔥 🔥 🔥 

I had to fork react-websocket and make a new component: [NextJS Websocket](https://www.npmjs.com/package/nextjs-websocket) 

I'm going to run tests on NextJS and normal React, but it should be up today or tomorrow 👍

**Comment by alamorre** [comment]
It works on NextJS and passed tests! You should be good on NextJS now with version *1.6.1* 😄 happy building

**Comment by JanKups** [comment]
@alamorre Thanks for the quick improvement!
